Gems/GradientSignal/Code/Source/ImageAsset.cpp

```diff
@@ -152,17 +151,17 @@ namespace GradientSignal
         return true;
     }
 
-    float GetValueFromImageAsset(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, const AZ::Vector3& uvw, float tilingX, float tilingY, float defaultValue)
+    float GetValueFromImageAsset(const AZ::Data::Asset<ImageAsset>& imageAsset, const AZ::Vector3& uvw, float tilingX, float tilingY, float defaultValue)
     {
         if (imageAsset.IsReady())
         {
-            const AZ::RHI::ImageDescriptor imageDescriptor = imageAsset->GetImageDescriptor();
-            auto width = imageDescriptor.m_size.m_width;
-            auto height = imageDescriptor.m_size.m_height;
+            const auto& image = imageAsset.Get();
+            AZStd::size_t imageSize = image->m_imageWidth * image->m_imageHeight *
+                static_cast<AZ::u32>(image->m_bytesPerPixel);
             
-            if (width > 0
-                && height > 0
-                )
+            if (image->m_imageWidth > 0 &&
+                image->m_imageHeight > 0 &&
+                image->m_imageData.size() == imageSize)
             {
```

Comment:
Nit because you didn't actually change functionality: This all seems redundant to me. They can't be less than 0, and if either were zero, then imageSize would be zero. It doesn't seem like ```image->m_imageData.size() == imageSize``` could ever be false. I could see maybe having a check for ```image->m_imageData.size() > 0``` but it's probably not even possible to produce 0 length image assets through normal means.

Reply:
Yeah I reverted those files since they'll be part of a follow-up task, but I will fix this up when I come back to this file! (which will be very soon)